Userland/Services/WindowServer/WindowFrame.cpp

```diff
@@ -399,16 +384,16 @@ void WindowFrame::render_to_cache()
 
     m_has_alpha_channel = Gfx::WindowTheme::current().frame_uses_alpha(window_state_for_theme(), WindowManager::the().palette());
 
-    static Gfx::Bitmap* s_tmp_bitmap;
+    static RefPtr<Gfx::Bitmap> s_tmp_bitmap;
```

Comment:
is this a static local variable? if so any chances to make it not one?

Reply:
What would prefer this to be instead? A file-level static variable?

Reply:
yeah file level static would probably be better.

Reply:
however you'd have to rename it to show what its intent is

Reply:
IMO having it as a function-level variable is better for the same reason why you wouldn't move other non-static variables into the file-level scope. Why is this better? :)

Reply:
isn't there quite a bit of overhead in static locals?

Reply:
Overhead? There's an atomic check and some futex action to initialize it initially, but after that it's just an atomic load to verify that it's been initialized already.

Reply:
Actually, it seems like the code is identical, except for the changed symbol name:

```Patch
[gunnar@nyx i686 (windowserver-bitmap-alloc)]$ diff -Nau x y
--- x	2021-06-02 13:54:55.355903237 +0200
+++ y	2021-06-02 13:55:33.259996408 +0200
@@ -830,7 +830,7 @@
 00061a40 l     F .text	00000207 WindowServer::WindowFrame::handle_menu_mouse_event(WindowServer::Menu&, WindowServer::MouseEvent const&) [clone .localalias]
 00061c50 l     F .text	000001c3 WindowServer::WindowFrame::handle_menubar_mouse_event(WindowServer::MouseEvent const&) [clone .localalias]
 00062bb0 l     F .text	000002c5 WindowServer::WindowFrame::paint_simple_rect_shadow(Gfx::Painter&, Gfx::Rect<int> const&, Gfx::Bitmap const&) const [clone .localalias]
-0008e5c4 l     O .bss	00000004 WindowServer::WindowFrame::render_to_cache()::s_tmp_bitmap
+0008e5c4 l     O .bss	00000004 WindowServer::s_tmp_bitmap
 00062e80 l     F .text	00000d4d WindowServer::WindowFrame::render_to_cache() [clone .localalias]
 0000b8c0 l     F .text	000000ca _GLOBAL__sub_I_WindowFrame.cpp
 0008d4ec l     O .data.rel.ro	0000000c typeinfo for AK::Function<void (WindowServer::Button&)>::CallableWrapper<WindowServer::WindowFrame::WindowFrame(WindowServer::Window&)::{lambda(auto:1&)#3}>
```

Reply:
i see, nevermind then.